packages/braintree-integration/src/braintree-visa-checkout/braintree-visa-checkout-customer-strategy.ts

```diff
@@ -0,0 +1,253 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+
+import {
+    BraintreeDataCollector,
+    BraintreeSdk,
+    BraintreeVisaCheckout,
+    VisaCheckoutAddress,
+    VisaCheckoutPaymentSuccessPayload,
+    VisaCheckoutTokenizedPayload,
+} from '@bigcommerce/checkout-sdk/braintree-utils';
+import {
+    Address,
+    CustomerInitializeOptions,
+    CustomerStrategy,
+    ExecutePaymentMethodCheckoutOptions,
+    InvalidArgumentError,
+    LegacyAddress,
+    MissingDataError,
+    MissingDataErrorType,
+    NotImplementedError,
+    PaymentIntegrationService,
+    RequestOptions,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import { WithBraintreeVisaCheckoutCustomerInitializeOptions } from './braintree-visa-customer-initialize-options';
+
+export default class BraintreeVisaCheckoutCustomerStrategy implements CustomerStrategy {
+    private buttonClassName = 'visa-checkout-wrapper';
+
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private formPoster: FormPoster,
+        private braintreeSdk: BraintreeSdk,
+    ) {}
+
+    async initialize(
+        options: CustomerInitializeOptions & WithBraintreeVisaCheckoutCustomerInitializeOptions,
+    ): Promise<void> {
+        const { braintreevisacheckout: visaCheckoutOptions, methodId } = options;
+
+        if (!visaCheckoutOptions || !methodId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "options.braintreevisacheckout" argument is not provided.',
+            );
+        }
+
+        await this.paymentIntegrationService.loadPaymentMethod(methodId);
+
+        const state = this.paymentIntegrationService.getState();
+
+        const checkout = state.getCheckout();
+
+        const { clientToken, config } = state.getPaymentMethodOrThrow(methodId);
+
+        const storeConfig = state.getStoreConfigOrThrow();
+
+        if (!checkout) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckout);
+        }
+
+        if (!storeConfig) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+        }
+
+        if (!clientToken) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
```

Comment:
It is not necessary to throw custom error if `checkout` or `storeConfig` is not defined, because `getStoreConfigOrThrow` or `getCheckoutOrThrow` methods make the same under the hood

```suggestion
        const state = this.paymentIntegrationService.getState();
        const checkout = state.getCheckoutOrThrow();
        const storeConfig = state.getStoreConfigOrThrow();
        const { clientToken, config } = state.getPaymentMethodOrThrow(methodId);

        if (!clientToken) {
            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
```